root/static/scripts/edit/URLCleanup.js

```diff
@@ -597,15 +597,28 @@ const CLEANUPS = {
     type: LINK_TYPES.downloadpurchase,
     clean: function (url) {
       url = url.replace(/^(?:https?:\/\/)?(?:(?:classic|pro|www)\.)?beatport\.com\//, "https://www.beatport.com/");
-      url = url.replace(/^(https:\/\/www\.beatport\.com)\/[\w-]+\/html\/content\/([\w-]+)\/0*([0-9]+)\/([\w-]+).*$/, "$1/$2/$4/$3");
-      url = url.replace(/^(https:\/\/www\.beatport\.com)\/[\w-]+\/html\/content\/([\w-]+)\/0*([0-9]+)(?:[\/?#].*)?$/, "$1/$2/-/$3");
-      url = url.replace(/^(https:\/\/www\.beatport\.com)\/([\w-]+)\/([\w-]+)\/0*([0-9]+).*$/, "$1/$2/$3/$4");
-      url = url.replace(/^(https:\/\/www\.beatport\.com)\/([\w-]+)\/\/+0*([0-9]+)(?:[\/?#].*)?$/, "$1/$2/-/$3");
-      url = url.replace(/^(https:\/\/www\.beatport\.com)\/([\w-]+)\/0*([0-9]+)(?:[\/?#].*)?$/, "$1/$2/$3");
+      var m = url.match(/^(https:\/\/www\.beatport\.com)\/[\w-]+\/html\/content\/([\w-]+)\/detail\/0*([0-9]+)\/([^\/?&#]+).*$/);
+      if (m) {
+        const slug = m[4].toLowerCase()
+          .replace(/%21/g, '-!-').replace(/(!+)/g, '-$1-')
+          .replace(/%23/g, '-pound-')
+          .replace(/%24/g, '-money-').replace(/\$/g, '-money-')
+          .replace(/%25/g, '-percent-')
+          .replace(/%26/g, '-and-')
+          .replace(/%40/g, '-at-').replace(/@/g, '-at-')
+          .replace(/%[0-9a-f]{2}/g, '-')
+          .replace(/%/g, '-percent-')
+          .replace(/[^a-z0-9!]/g, '-')
+          .replace(/-+/g, '-')
+          .replace(/^-|-$/g, '')
+          .replace(/^-?$/g, '---');
```

Comment:
Not sure what these last three replacements are doing, but I'll hope they're covered by some test already. :)

Reply:
It strip extraneous hyphens and replace empty slug with 3 hyphens (see track [?????? ??????](https://www.beatport.com/release/---/1990510) for example). They are fully covered by the new made up example.